sydent/validators/msisdnvalidator.py

```diff
@@ -44,20 +70,43 @@ def requestToken(self, msisdn, clientSecret, sendAttempt, nextLink):
             logger.info("Not texting code because current send attempt (%d) is not less than given send attempt (%s)", int(sendAttempt), int(valSession.sendAttemptNumber))
             return valSession.id
 
+        smsBodyTemplate = self.sydent.cfg.get('sms', 'bodyTemplate')
+        originator = self.getOriginator(phoneNumber)
+
         logger.info(
-            "Attempting to text code %s to %s",
-            valSession.token, msisdn,
+            "Attempting to text code %s to %s (country %d) with originator %s",
+            valSession.token, msisdn, phoneNumber.country_code, originator
         )
 
-        smsBodyTemplate = self.sydent.cfg.get('sms', 'bodyTemplate')
-
         smsBody = smsBodyTemplate.format(token=valSession.token)
 
-        self.omSms.sendTextSMS(smsBody, msisdn)
+        self.omSms.sendTextSMS(smsBody, msisdn, originator)
 
         valSessionStore.setSendAttemptNumber(valSession.id, sendAttempt)
 
         return valSession.id
 
+    def getOriginator(self, destPhoneNumber):
+        countryCode = str(destPhoneNumber.country_code)
+
+        origs = [{
+            "type": "alpha",
```

Comment:
The TONS constant is for looking up the strings to the numbers - to eliminate the duplicated strings we could define extra constants for the strings and import them, but it didn't really seem worth it.